src/algebra/module/composition_series.lean

```diff
@@ -0,0 +1,107 @@
+import order.jordan_holder
+import order.cover
+import ring_theory.simple_module
+
+def partial_order.bot_or_top {α : Type*} [partial_order α] (x z : α) :=
+  ∀ ⦃y⦄, x ≤ y → y ≤ z → (y = x ∨ y = z)
+
+lemma covby.bot_or_top {α : Type*} [partial_order α] {x z : α} (hxz : x ⋖ z) :
+  partial_order.bot_or_top x z := λ y hxy hyz,
+begin
+  cases lt_or_eq_of_le hxy with hxy h,
+  cases lt_or_eq_of_le hyz with hyz h,
+  { exfalso,
+    exact hxz.2 hxy hyz },
+  exact or.inr h,
+  exact or.inl h.symm
+end
+
+lemma covby.iff_bot_or_top {α : Type*} [partial_order α] {x z : α} :
+  x ⋖ z ↔ (x < z ∧ partial_order.bot_or_top x z ) :=
+begin
+  split; intro h,
+  exact ⟨h.lt, h.bot_or_top⟩,
+  refine ⟨h.1, _⟩,
+  intros y hxy hyz,
+  apply lt_irrefl y,
+  cases h.2 (le_of_lt hxy) (le_of_lt hyz) with h' h';
+    rw ←h' at *; assumption
+end
+
+lemma covby.sup_eq {α : Type*} [lattice α] {x y z : α} : x ⋖ z → y ⋖ z → x ≠ y → x ⊔ y = z :=
```

Comment:
I think this might be true for `wcovby`

Reply:
That sounds right. `x ⊔ y ≤ z` because `x ≤ z` and `y ≤ z`, and if `x ⊔ y < z` (`has_le.le.eq_of_not_lt`) we would contradict either `x ⋖ z` or `y ⋖ z` because `x ⊔ y` is greater than one of `x` or `y` (would be great to have `ne.lt_sup_or_lt_sup`).

Reply:
#17274